### PR TITLE
infra: revert "Update GPU kernel for Decision Forest (#2402)"

### PR DESCRIPTION
**Pull request by ethanglaser**
Reverts https://github.com/oneapi-src/oneDAL/commit/699774b81c34baec0ca0d68fa3af6f81f9eb74fc

### CODE REVIEW

**Review by samir-nasibli — APPROVED** [review]

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl.hpp [hunk 1]
```diff
@@ -627,8 +632,10 @@ class train_kernel_hist_impl {
...
 
     pr::ndarray<Float, 1> res_var_imp_;
+
+    node_group_list_t node_group_list_;
 };
 
 #endif
 
-} // namespace oneapi::dal::decision_forest::backend
+} // namespace oneapi::dal::decision_forest::backend
```
Comment (samir-nasibli) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl.hpp:641]: please add also blank line at the EOF

**Comment by ethanglaser** [comment]
CI job: http://intel-ci.intel.com/ee36d683-5179-f10f-b9eb-a4bf010d0e2e

**Pull request merged**